packages/x-components/src/components/base-variable-column-grid.vue

```diff
@@ -55,7 +55,26 @@
      *
      * @internal
      */
-    protected columns = 0;
+    @Prop({ default: 0 })
+    protected columns!: number;
+
+    /**
+     * The number of columns provided by the user.
```

Comment:
Can we change to `provided by a user interaction`? For me, it's clearer that way